src/tests/test_kube_api.py

```diff
@@ -83,14 +75,344 @@ def _get_vips(cls, cluster_config, nodes: Nodes):
+
+    @JunitTestSuite()
+    @pytest.mark.kube_api
+    def test_kube_api_ipv6(self, kube_test_configs_single_node, kube_api_context, proxy_server, get_nodes):
+        cluster_config, tf_config = kube_test_configs_single_node
+        tf_config.is_ipv6 = True
+        tf_config.is_ipv4 = False
+
+        self.kube_api_test(
+            kube_api_context.api_client,
+            get_nodes(tf_config, cluster_config),
+            cluster_config,
+            global_variables.spoke_namespace,
+            proxy_server,
+            is_ipv4=False,
+        )
+
+    @classmethod
+    def start_nodes(cls, nodes: Nodes, infra_env: InfraEnv, entity_config: BaseEntityConfig, is_ipv4) -> List[Agent]:
+        infra_env.status()
+        cls.download_iso_from_infra_env(infra_env, entity_config.iso_download_path)
+
+        log.info("iso downloaded, starting nodes")
+        nodes.start_all()
+
+        log.info("waiting for host agent")
+        agents = infra_env.wait_for_agents(len(nodes))
+        for agent in agents:
+            agent.approve()
+            cls.set_agent_hostname(nodes[0], agent, is_ipv4)  # Currently only supports single node
+
+        return agents
+
+    def kube_api_test(
+        self,
+        api_client: ApiClient,
+        nodes: Nodes,
+        cluster_config: ClusterConfig,
+        spoke_namespace: str,
+        proxy_server: Optional[Callable] = None,
+        *,
+        is_ipv4: bool = True,
+        is_disconnected: bool = False,
+    ):
+        cluster_name = cluster_config.cluster_name.get()
+
+        # TODO resolve it from the service if the node controller doesn't have this information
+        #  (please see cluster.get_primary_machine_cidr())
+        machine_cidr = nodes.controller.get_primary_machine_cidr()
+
+        agent_cluster_install = AgentClusterInstall(
+            api_client, f"{cluster_name}-agent-cluster-install", spoke_namespace
+        )
+
+        secret = Secret(api_client, f"{cluster_name}-secret", spoke_namespace)
+        secret.create(pull_secret=cluster_config.pull_secret)
+
+        cluster_deployment = ClusterDeployment(api_client, cluster_name, spoke_namespace)
+        cluster_deployment.create(agent_cluster_install_ref=agent_cluster_install.ref, secret=secret)
+
+        agent_cluster_install.create(
+            cluster_deployment_ref=cluster_deployment.ref,
+            image_set_ref=self.deploy_image_set(cluster_name, api_client),
+            cluster_cidr=cluster_config.cluster_networks[0].cidr,
+            host_prefix=cluster_config.cluster_networks[0].host_prefix,
+            service_network=cluster_config.service_networks[0].cidr,
+            ssh_pub_key=cluster_config.ssh_public_key,
+            hyperthreading=cluster_config.hyperthreading,
+            control_plane_agents=nodes.masters_count,
+            worker_agents=nodes.workers_count,
+            machine_cidr=machine_cidr,
+        )
+        agent_cluster_install.wait_to_be_ready(False)
```

Comment:
```suggestion
        agent_cluster_install.wait_to_be_ready(ready=False)
```

it's still a bit weird, but at least more clear as to what should happen

Reply:
Or maybe ``wait_to_be_ready`` should be changed to ``wait_for_status`` or ``wait_to_state``

Reply:
I'm actually not sure we need `wait_to_be_ready` function. It just calls `wait_for_condition` with 2 extra parameters (timeout is the default anyway).
The only think I don't understand is why when calling to `wait_for_condition` we `str(ready : bool)`

Reply:
I think that currently I'll do as you suggested `agent_cluster_install.wait_to_be_ready(ready=False)` and will open a ticket for cleanup kubeapi infra code